syz-ci/manager.go

```diff
@@ -242,17 +242,18 @@ loop:
 
 func (mgr *Manager) pollAndBuild(lastCommit string, latestInfo *BuildInfo) (
 	string, *BuildInfo, time.Duration) {
-	var success bool
 	rebuildAfter := buildRetryPeriod
 	commit, err := mgr.repo.Poll(mgr.mgrcfg.Repo, mgr.mgrcfg.Branch)
 	if err != nil {
+		mgr.buildFailed = true
 		mgr.Errorf("failed to poll: %v", err)
 	} else {
 		log.Logf(0, "%v: poll: %v", mgr.name, commit.Hash)
-		if commit.Hash != lastCommit &&
-			(latestInfo == nil ||
-				commit.Hash != latestInfo.KernelCommit ||
-				mgr.configTag != latestInfo.KernelConfigTag) {
+		needsUpdate := (latestInfo == nil ||
+			commit.Hash != latestInfo.KernelCommit ||
+			mgr.configTag != latestInfo.KernelConfigTag)
+		mgr.buildFailed = needsUpdate
```

Comment:
Or we do first build before we start manager?

Reply:
> We can temporary set it to true even if the build will succeed, but this is probably fine, right?

It should not have any effect until we have exited the function and by that time it will already have the correct value. All poll/build/restart logic is happening in a single loop, so there should be no restarts in the meanwhile.

> Or we do first build before we start manager?

We don't do it if the kernel image is fresh enough.

https://github.com/google/syzkaller/blob/1c0ecc51adc9f44c7bd3f45c2aa1b62718d1236e/syz-ci/manager.go#L179-L182